### ISSUE TITLE
[BUG] pyximport is ignoring `sys.path`

### ISSUE THREAD

**Issue by matusvalo** [cython/cython#5615]
## Describe your issue

Consider following file tree:
```cython
########## mydir/mytest.pyx ##########
print('Hello world!')
```

Running command:

```sh
PYTHONPATH=mydir python -c 'import pyximport; pyximport.install(); import mytest'
```
from parent directory of `mydir` will fail with following error:
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'mytest'
```

## Expected behaviour

Compilation success since the module is present in `PYTHONPATH`.

## OS
any

## Python version
any

## Cython version
master

## Additional context
This bug is regression. Cython 0.29.X works correctly.

**Comment by scoder** [comment]
I'm not entirely convinced that we should change the behaviour of 3.0.x at this late stage. If it wasn't like this in 3.0.0, why change it it 3.0.12? We can easily change it for 3.1, but what is the reason for backport it?

**Comment by matusvalo** [comment]
> If it wasn't like this in 3.0.0, why change it it 3.0.12?

If I am not wrong, the behaviour of pyximport in 3.0.0 changed comparing to old 0.29.X. Hence, it this bug should be regression and potential candidate for fix in 3.0.X I suppose.

**Comment by scoder** [comment]
> A workaround hack is to import from the old version _pyximport2 for all python versions

Actually, this is a good argument. It means that there is a difference between the Python 2 and Python 3 versions of pyximport that is unrelated to the runtime differences.

A backport is ok then.
